src/AppInstallerRepositoryCore/Microsoft/Schema/ISQLiteIndex.h

```diff
@@ -108,6 +108,8 @@ namespace AppInstaller::Repository::Microsoft::Schema
         virtual std::set<std::pair<SQLite::rowid_t, Utility::NormalizedString>> GetDependenciesByManifestRowId(const SQLite::Connection& connection, SQLite::rowid_t manifestRowId) const = 0;
 
         virtual std::vector<std::pair<SQLite::rowid_t, Utility::NormalizedString>> GetDependentsById(const SQLite::Connection& connection, AppInstaller::Manifest::string_t packageId) const = 0;
+
+        virtual bool NotNeeded(const SQLite::Connection& connection, std::string_view tableName, std::string_view valueName, SQLite::rowid_t id) const = 0;

```

Comment:
This doesn't need to be public, so it can be a `protected` function on the base class rather than on the interface.